SwordImportExportPlugin.inc.php

```diff
@@ -0,0 +1,230 @@
+					$request, ROUTE_PAGE,
+					null, 'management', 'settings', 'website',
+					array(),
+					'swordSettings'
+				);
+				foreach (array('selectedDepositPoint', 'depositEditorial', 'depositGalleys') as $var) {
+					$templateMgr->assign($var, $request->getUserVar($var));
+				}
+				$templateMgr->assign('swordSettingsPageUrl', $settingUrl);
+				$templateMgr->assign('depositPoints', $depositPointsData);
+				$templateMgr->assign('pluginJavaScriptURL', $this->getSwordPlugin()->getJsUrl($request));
+				$templateMgr->display($this->getTemplatePath() . 'articles.tpl');
+				break;
+
+			case 'deposit':
+				$context = $request->getContext();
+				$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+				$this->getSwordPlugin()->import('classes.OJSSwordDeposit');
+
+				$depositPointId = $request->getUserVar('depositPoint');
+				$password = $request->getUserVar('swordPassword');
+				if ($password == SWORD_PASSWORD_SLUG) {
+					$depositPointDao = DAORegistry::getDAO('DepositPointDAO');
+					$depositPoint = $depositPointDao->getById($depositPointId, $context->getId());
+					if ($depositPoint) {
+						$password = $depositPoint->getSwordPassword();
+					}
+				}
+
+				$swordDepositPoint = $request->getUserVar('swordDepositPoint');
+				$depositEditorial = $request->getUserVar('depositEditorial');
+				$depositGalleys = $request->getUserVar('depositGalleys');
+				$username = $request->getUserVar('swordUsername');
+				$depositIds = array();
+
+				$backLink = $request->url(
+					null, null, null,
+					array('plugin', $this->getName()),
+					array(
+						'selectedDepositPoint' => $depositPointId,
+						'depositEditorial' => $depositEditorial,
+						'depositGalleys' => $depositGalleys,
+					)
+				);
+
+				$errors = array();
+				// select at least one article
+				$articleIds = $request->getUserVar('articleId');
+				if (empty($articleIds)) {
+					$errors[] = array(
+						'title' => __('plugins.importexport.sword.requiredFieldErrorTitle'),
+						'message' => __('plugins.importexport.sword.requiredFieldErrorMessage'),
+					);
+				}
+				else {
+					foreach ($articleIds as $articleId) {
+						$publishedArticle = $publishedArticleDao->getByArticleId($articleId);
+						try {
+							$deposit = new OJSSwordDeposit($publishedArticle);
+							$deposit->setMetadata($request);
+							if ($depositGalleys) $deposit->addGalleys();
+							if ($depositEditorial) $deposit->addEditorial();
+							$deposit->createPackage();
+							$response = $deposit->deposit($swordDepositPoint, $username, $password);
+							$deposit->cleanup();
+							$depositIds[] = $response->sac_id;
+						}
+						catch (Exception $e) {
+							$errors[] = array(
+								'title' => $publishedArticle->getLocalizedTitle(),
+								'message' => $e->getMessage(),
+							);
+						}
+					}
+				}
+
+				if (!empty($errors)) {
+					$errorMessage = '<dl>';
+					foreach ($errors as $error) {
+						$errorMessage .= "<dt>{$error['title']}</dt>";
```

Comment:
The strings will need HTML escaping, I think. If possible, it's better to do your formatting in the template rather than mixing markup into your PHP.

Reply:
fixed!